osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -394,6 +394,15 @@ private void addChannel(Channel channel)
             {
                 careChannels.Add(channel);
                 channelTabs.AddItem(channel);
+
+                if (channel.Type == ChannelType.Public && !channel.Joined)
+                {
+                    var req = new JoinChannelRequest(channel, api.LocalUser);
+                    req.Success += () => addChannel(channel);
+                    req.Failure += ex => removeChannel(channel);
```

Comment:
I think it should be logged if the request fails.

Reply:
It is inherently logged by the network subsystem. UI messaging will be added later.